### PR TITLE
Add giveaway endpoints

### PR DESCRIPTION
**Pull request by marciammart**
- Create endpoints to list, create and view giveaways in detail
- Add endpoint tests
- Add documentation and yarn script to help setup

### CODE REVIEW

File: packages/node-app/src/models/giveaway.model.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  unit?: Unit;
+}
+
+interface Giveaway extends mongoose.Document {
+  title: string;
+  description: string;
+  startTime: Date;
+  endTime: Date;
+  participants: string[];
+  winners: string[];
```
Comment (goncaloadolfo) [inline packages/node-app/src/models/giveaway.model.ts:19]: I think we should also have a user schema, so we can map the email with the name, role, profile image, etc

File: packages/node-app/src/controllers/giveaway.controller.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  try {
+    const { title, description, startTime, endTime, numberOfWinners, requirements } = req.body;
+    const giveaway = new Giveaway({
+      title,
+      description,
+      startTime: new Date(startTime),
+      endTime: new Date(endTime),
+      numberOfWinners,
+      requirements
+    });
```
Comment (goncaloadolfo) [inline packages/node-app/src/controllers/giveaway.controller.ts:34]: Shouldn't we perform some validations on the input? For example, the end time > start time, and negative number of winners. I know that this validation is in the contract, but in these case I think it should return a bad request instead

File: packages/node-app/src/tests/giveaway.test.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+    const response = await request(app).get(`/giveaways/${giveaway._id}`);
+    expect(response.status).toBe(200);
+    expect(response.body.title).toBe('Giveaway');
+    expect(response.body.description).toBe('Description for giveaway');
+    expect(new Date(response.body.startTime)).toEqual(giveaway.startTime);
+    expect(new Date(response.body.endTime)).toEqual(giveaway.endTime);
+    expect(response.body.numberOfWinners).toEqual(1);
+  });
+});
```
Comment (goncaloadolfo) [inline packages/node-app/src/tests/giveaway.test.ts:85]: I think there is a case missing to post a giveaway

File: packages/node-app/src/models/giveaway.model.ts (hunk 1 above)
Comment (marciammart) [reply packages/node-app/src/models/giveaway.model.ts:19]: We will get the user information from an external database so the only thing we need to save on our side is a list of the account addresses that will be used to identify the users

**Review by goncaloadolfo — APPROVED** [review]
LGTM, good job! 

Just one point that we should consider in the future:
- posting giveaways with a past end date

For now I would say to move forward with the PR, this can be handled separately.

**Pull request merged**